packages/app/obojobo-document-engine/src/scripts/common/util/trigger-util.js

```diff
@@ -0,0 +1,96 @@
+/*
+	Mutates triggers
+	triggerMap is keyed by trigger type
+	with a value of an action (object)
+	or multiple actions (array of objects)
+	Example triggerMap:
+		const triggerMap = {
+			onStartAttempt: { type: 'nav:lock', someValue: 3 },
+			onEndAttempt: [{ type: 'nav:unlock' }, {type: 'someType'}]
+		}
+*/
+export const addActionsToTriggers = (triggers, triggerMap) => {
+	const usesMap = { ...triggerMap }
+	for (const type in usesMap) {
+		usesMap[type] = false
+	}
+
+	for (const trigger of triggers) {
+		let actions = triggerMap[trigger.type]
+		if (actions) {
+			if (!trigger.actions) trigger.actions = []
+			// if just an object convert to array
+			if (actions.constructor.name === 'Object') actions = [actions]
```

Comment:
I like using typeof, like `typeof actions === 'object'` - would that work or is constructor.name considered safer/more accurate?

Reply:
Actually, what about...

```suggestion
			actions = [].concat(actions)
```

I think that should work both if actions is already an array or if it's not